bundle/filefs.go

```diff
@@ -0,0 +1,66 @@
+// +build go1.16
+
+package bundle
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+
+	"github.com/pkg/errors"
+)
+
+type dirLoaderFS struct {
+	filesystem fs.FS
+	files      []string
+	idx        int
+}
+
+// NewFSLoader returns a basic DirectoryLoader implementation
+// that will load files from a fs.FS interface
+func NewFSLoader(filesystem fs.FS) DirectoryLoader {
+	d := dirLoaderFS{
+		filesystem: filesystem,
+	}
+
+	return &d
+}
+
+// NextFile iterates to the next file in the directory tree
+// and returns a file Descriptor for the file.
+func (d *dirLoaderFS) NextFile() (*Descriptor, error) {
+	// build a list of all files we will iterate over and read, but only one time
+	if d.files == nil {
+		err := fs.WalkDir(d.filesystem, ".", func(path string, dirEntry fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if dirEntry != nil && dirEntry.Type().IsRegular() {
+				d.files = append(d.files, path)
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to list files")
```

Comment:
Sorry, I've overlooked this. Since this is new code, let's do
```suggestion
			return nil, fmt.Errorf("failed to list files: %w", err)
```
instead